ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -170,8 +168,8 @@ func (scbuilder *SiteConfigBuilder) getWorkloadManifest(cpuSet string) (string,
 }
 
 func (scbuilder *SiteConfigBuilder) getExtraManifest(dataMap map[string]interface{}, clusterSpec Clusters) (map[string]interface{}, error) {
```

Comment:
Rather than passing dataMap in as a parameter and then returning the updated map (all resulting in copies of the map) you can pass dataMap in as a pointer and modify in place. Just be sure that you handle the error case in the caller if you are relying on returning "nil" to clear it out.

Reply:
not really fan of using pointers especially as there is a recursive call function to get the field value.